### PR TITLE
Add a Test environment 

### PR DESCRIPTION
**Pull request by henrykironde**

### CODE REVIEW

File: Zooniverse/environment_mbtiles.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: mbtilesEnv
```
Comment (ethanwhite) [inline Zooniverse/environment_mbtiles.yml:1]: Capitalization here `mbtilesEnv` is different from that used in the rules `mbtilesenv`. We should make these consistent even if it works.

File: everglades_dryrun_workflow.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+#SBATCH --job-name=everglades_workflow 
+#SBATCH --mail-user=henrysenyondo@ufl.edu 
+#SBATCH --mail-type=FAIL
+#SBATCH --gpus=a100:4
+#SBATCH --cpus-per-task=5
+#SBATCH --mem=1000gb
+#SBATCH --time=180:00:00
```
Comment (ethanwhite) [inline everglades_dryrun_workflow.sh:8]: We should need much less time from this so suggest scaling back to a reasonable number to help with scheduling.

File: everglades_workflow.sh [hunk 3]
```diff
@@ -3,9 +3,9 @@
 #SBATCH --mail-user=henrysenyondo@ufl.edu 
 #SBATCH --mail-type=FAIL
 #SBATCH --gpus=a100:4
-#SBATCH --cpus-per-task=10
```
Comment (ethanwhite) [inline everglades_workflow.sh:6]: Is there a reason we're making this number smaller?

File: everglades_workflow.sh [hunk 4]
```diff
@@ -16,9 +16,10 @@ echo "INFO [$(date "+%Y-%m-%d %H:%M:%S")] Loading required modules"
 source /etc/profile.d/modules.sh
 
 ml conda
-conda env create --force --file /blue/ewhite/everglades/EvergladesTools/Zooniverse/environment.yml
```
Comment (ethanwhite) [inline everglades_workflow.sh:19]: Delete instead of comment out?

**Review by ethanwhite — CHANGES_REQUESTED** [review]
This all looks good (though I skimmed over a lot of repetitive changes. The one thing I wonder about is the large duplication of the following code:

```python
    test_env_name = "TEST_ENV"
    test_env_set = os.environ.get(test_env_name)
    working_dir = "/blue/ewhite/everglades_test" if test_env_set else "/blue/ewhite/everglades"
```

I'd suggest one of two approaches to reducing that:
1. Pass `working_dir` to each of the functions as an argument. We already determine it in the Snakefile, so add an additional argument to pass it to the functions instead of rewriting the code.
2. Write a function that contains the above code so that each module can just run `working_dir = get_working_dir()`.
3. Infer it from the information already passed to the functions. Snakemake is passing in paths to each of the functions already, take the first path and check the working dir from it (should be a single line of code).

I think (1) is probably the cleanest, but let me know what you think (or if there's some complexity I'm missing here).

File: everglades_workflow.sh (hunk 3 above)
Comment (henrykironde) [reply everglades_workflow.sh:6]: I think PyTorch has some issues with the number of workers. Ben suggested reducing that number. I can give it a test after our current job finishes today.

File: everglades_dryrun_workflow.sh (hunk 2 above)
Comment (henrykironde) [reply everglades_dryrun_workflow.sh:8]: I'll trim it to 72 hours, roughly the time elapsed for the last two runs of a clean pipeline, which was about 3 days. I'll further reduce it pending how playing with number of workers goes.

Comment (ethanwhite) [reply everglades_dryrun_workflow.sh:8]: This is a dry run file though. I assume the test setup is running much faster right?

File: everglades_workflow.sh (hunk 4 above)
Comment (ethanwhite) [reply everglades_workflow.sh:19]: Yeah, in the docs sounds better. Speaking of which, I was going to note that we should add a description of this setup to the README as part of the PR.

File: Zooniverse/process_nests.py [hunk 5]
```diff
@@ -105,9 +105,14 @@ def process_nests(nest_file, year, site, savedir, min_score=0.3, min_detections=
 
 
 if __name__ == "__main__":
+    # Check if the test environment variable exists
+    test_env_name = "TEST_ENV"
+    test_env_set = os.environ.get(test_env_name)
+    working_dir = "/blue/ewhite/everglades_test" if test_env_set else "/blue/ewhite/everglades"
```
Comment (ethanwhite) [inline Zooniverse/process_nests.py:111]: Looks like one more chunk to update to the new function

**Review by ethanwhite — CHANGES_REQUESTED** [review]
One more chunk of test env checking to update to the function.

**Pull request merged**